jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/ctl/ActionCompact.java

```diff
@@ -88,18 +88,25 @@ private static DatasetGraph unwrap(DatasetGraph dsg, Predicate<DatasetGraph> pre
     }
 
     static class CompactTask extends TaskBase {
-        static private Logger log = Fuseki.compactLog;
+        static private final Logger log = Fuseki.compactLog;
+
+        private final boolean shouldDeleteOld;
 
         public CompactTask(HttpAction action) {
             super(action);
+
+            String deleteOldParam = action.request.getParameter("deleteOld");
+
+            this.shouldDeleteOld = ( deleteOldParam != null
+                && ( deleteOldParam.isEmpty() || deleteOldParam.equalsIgnoreCase("true") ) );
```

Comment:
If I understood it correctly (having first :coffee: of the day), in the command line the database is not deleted by default.

But here in Fuseki if you do not pass a parameter (`http//url/`?) then it is not deleted. If you pass an empty parameter (`http//url/?deleteOld=` ?) then it is deleted? Or if you pass `"true"` (`http//url/?deleteOld=true` ?) then it's also deleted.

If that's correct, then I think it's the old database is not deleted only if I do not specify the parameter or if I set its value to `"false"`? Shouldn't we ignore blank/empty strings too, and only delete if the param is `true`, having this way the same behaviour as in the cli?

```suggestion
            this.shouldDeleteOld = deleteOldParam != null && deleteOldParam.trim().equalsIgnoreCase("true");
```

Reply:
The reason that I coded it this way is so that the following will both trigger a deletion: `http//url/?deleteOld` & `http//url/?deleteOld=true`. Since `deleteOld` is a boolean/flag parameter, its mere presence implies `true` if no explicit value is set. This is pretty common for boolean URL params in my experience.

Reply:
I actually believe that this probably should be in the body of the POST, but I'm trying to follow already existing conventions in the project.

Reply:
Ah, now that makes sense @bsara. I was wondering why you'd chose a different behavior here :+1: thanks!

Reply:
Makes sense but we should ensure that this is explicit in the documentation